filter/filter.go

```diff
@@ -31,13 +35,55 @@ func (f *Filter) IsExcluded(v string) bool {
 	return ok
 }
 
+// Validate validates that the data inside of the filters is right
+func (f *Filter) Validate() error {
+	// Validate that the Targets have the right format
+	if len(f.Targets) != 0 {
+		for _, t := range f.Targets {
+			split := strings.Split(t, ".")
+			// IDs can have . in between so we validate that at least we have the minimum
+			if len(split) < 2 {
+				return errors.Wrapf(errcode.ErrFilterTargetsInvalid, "the Target %q has an invalid format. The expected format is 'aws_instance.ID'", t)
+			}
+		}
+	}
+
+	return nil
+}
+
+// TargetsTypesWithIDs returns all the types (ex: aws_instance) from
+// the list of Targets and the IDs
+func (f *Filter) TargetsTypesWithIDs() map[string][]string {
+	types := make(map[string]map[string]struct{})
+	res := make(map[string][]string)
+
+	for _, t := range f.Targets {
+		split := strings.Split(t, ".")
+		ty := split[0]
+		id := strings.Join(split[1:], ".")
+
+		if _, ok := types[ty]; !ok {
```

Comment:
Would we have a method to directly validate that the type exist in terraform as well?
e.g.: `aws_intsance` has a typo instead of `aws_instance`, this way we could discard incorrect type or report errors directly (as done in `import.go` via `if !p.HasResourceType(k) {` that would avoid iterrating over them again).

Reply:
It's already being validated https://github.com/cycloidio/terracognita/pull/80/files#diff-bb6bf40d52cce0b6a4a057a5cc4853ecR37

Reply:
Yeah, but why not do it directly there and avoid extra validation later on?